packages/discovery-react-components/src/utils/document/processDoc.ts

```diff
@@ -250,6 +251,14 @@ function setupSectionParser(
     },
 
     ontext: (_: Parser, text: string): void => {
+      if (decodeHTML(text).length != text.length) {
+        const lastElem = sectionHtml.length - 1;
```

Comment:
I think this is wrong. Since you're making changes to the _open tag_, do same as in `onclosetag()` below: get the index from `openTagIndices`.